### PR TITLE
VIH-7760 - changes done to fix retry of failed confirmation booking

### PR DESCRIPTION
**Pull request by khizarkhanu-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7760

### Change description ###

changes made to fix timing issue in higher environments, if comfirmation fails, it should not raise a reload event rather set the status in local storage.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts [hunk 1]
```diff
@@ -223,7 +223,11 @@ export class BookingDetailsComponent implements OnInit, OnDestroy {
         if (status === UpdateBookingStatus.Cancelled) {
             this.showCancelBooking = false;
         }
+        this.hearing.Status = UpdateBookingStatus[status];
```
Comment (robertbennett1998) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts:226]: Can this not just be:
`this.hearing.Status = status`
as status is already the correct type.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts [hunk 2]
```diff
@@ -223,7 +223,11 @@ export class BookingDetailsComponent implements OnInit, OnDestroy {
         if (status === UpdateBookingStatus.Cancelled) {
```
Comment (robertbennett1998) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts:223]: Could you also add a unit test to make sure `this.videoHearingService.getHearingById(this.hearingId)` is never called if the status is failed.

**Review by robertbennett1998 — CHANGES_REQUESTED** [review]
Overall this is fine and will fix the issue but it seems like there is a bigger underlying cause. This issue shouldn't exist anyways and suggests that there may need to be some refactoring.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts (hunk 1 above)
Comment (khizarkhanu-hmcts) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts]: I have now fixed to handle when status is failed so other unit test are not impacted.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts (hunk 2 above)
Comment (khizarkhanu-hmcts) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts]: it is already there and this is not called, pls refer to test 'should show pop up if the confirm failed' in booking-details.component.spec.ts

**Review by robertbennett1998 — APPROVED** [review]
LGTM! Previous comment still applies.

**Pull request merged**